azblob/parsing_urls.go

```diff
@@ -124,6 +124,11 @@ func (up BlobURLParts) URL() url.URL {
 
 	rawQuery := up.UnparsedParams
 
+	//If no snapshot is initially provided, fill it in from the SAS query properties to help the user
+	if up.Snapshot == "" && !up.SAS.snapshotTime.IsZero() {
+		up.Snapshot = up.SAS.snapshotTime.Format(SnapshotTimeFormat)
```

Comment:
FYI @rickle-msft @jaschrep-msft I remember we made the decision not to append the snapshot ID when the customer is asking for a SAS token. But in the Go SDK, the customer never asks for just the token, instead they ask for the whole URL. So here we help them by including the snapshot ID if they forgot to set it on the BlobURLParts.

Please let us know if you have any concern.

Reply:
I think this is fine. If they have a SAS that only goes to a snapshot, it seems reasonable that would help them towards targeting the resource they can actually access. I don't imagine anyone will complain that we didn't append their snapshot sas to a base blob and let the request fail.